src/components/PopupWithForm.js

```diff
@@ -20,8 +30,7 @@ class PopupWithForm extends Popup {
   setEventListeners() {
     this._popupForm.addEventListener("submit", (e) => {
       e.preventDefault();
-      const formValues = this._getInputValues();
-      this._handleFormSubmit(formValues);
+      this._handleFormSubmit(this._getInputValues());
       this._popupForm.reset();
```

Comment:
Needs correction: Reset the form in the case of successful api response in `then` block in `index.js`, otherwise the form will be empty even if an error occurred and you will need to fill the values again.

Reply:
Hey there! Thank you for your reviews. I just wanted to clarify on this one. When you mention "reset the form", does my resetValidation() not already do that? I guess I hadn't used the resetValidation in my handleChangeAvatar method, but I had used it in my other ones. I just wanted to clarify. Thank you for your time!

Reply:
Reset validation resets the validation state (the errors and the button state), not the form itself

In this comment I mean that you should not call `this._popupForm.reset();` here, do that only after the successful submission in index.js in `then` block, so the form won't be cleared in the card of an error

Reply:
Thank you for the clarification! Hopefully everything should look better now. Thank you for taking the time to review this again.